src/os/bluestore/BlueStore.cc

```diff
@@ -13255,7 +13304,25 @@ void BlueStore::_kv_sync_thread()
   timespan twait = ceph::make_timespan(0);
   size_t kv_submitted = 0;
 
+  ceph::timespan max_interval = std::chrono::microseconds(kv_sync_interval);
+  auto last = t0;
+  auto cur = t0;
+
   while (true) {
+    cur = mono_clock::now();
+    ceph::timespan sync_interval = cur - last;
+    if (sync_interval < max_interval && kv_queue.size() < kv_sync_items) {
+      continue;
+    }  else {
+      dout(20) << __func__ 
+              << ", kv_sync_interval: " << kv_sync_interval
+              << ", sync_interval: " << sync_interval
+              << ", kv_sync_items: " << kv_sync_items
+              << ", kv_queue.size(): " << kv_queue.size()
+              << dendl;
+      last = cur;
```

Comment:
Not sure I understand why you favor the first awaken op and allow it to go ahead without waiting for additional ones. IMO this might reduce the intended positive effect from your patch - part of ops wouldn't benefit from batching.. In the worst case when configured interval is below kv_sync_thread iteration duration one might get the case when batching isn't applied on each step: e.g. if it takes 10ms for the loop to complete op submissions and do all the other tasks - while configured interval for batching is 8ms - every next op from the queue would omit batching over and over...